app/api/helpers/user.py

```diff
@@ -0,0 +1,47 @@
+from sqlalchemy.orm.exc import NoResultFound
+from app.models import db
+from app.api.helpers.db import save_to_db
+from app.models.user import User
+
+
+def modify_email_for_user_to_be_deleted(user):
+    """
+    Update the email ID of user which is to be deleted.
+    Adds '.deleted' substring to email of a user to be deleted.
+    :param order: User to be deleted.
+    :return:
+    """
+    not_unique_email = True
+    user_email = user.email
+    while not_unique_email:
+        try:
+            db.session.query(User).filter_by(email=user_email).one()
+        except NoResultFound:
+            user.email = user_email
+            save_to_db(user)
+            not_unique_email = False
+        else:
+            user_email = user_email + '.deleted'
+    return user
+
+
+def modify_email_for_user_to_be_restored(user):
+    """
+    Update the email ID of user which is to be restored.
+    Removes '.deleted' substring from a user to be restored.
+    :param order: User to be restored.
+    :return:
+    """
+    not_unique_email = True
+    user_email = user.email
+    while not_unique_email:
+        try:
+            db.session.query(User).filter_by(email=user_email).one()
+        except NoResultFound:
+            user.email = user_email
+            save_to_db(user)
+            not_unique_email = False
+        else:
+            if user_email.endswith('.deleted'):
+                user_email = user_email[:-8]
```

Comment:
Why do we need this?

Reply:
When the user is deleted, `.deleted` is added so when the deleted user is restored, the original email should be restored which is substring obtained by omitting last 8 characters i.e. `.deleted` @iamareebjamal

Reply:
Not maintainable. Remove the loop. Remove '.deleted' from the last. If you have to use :-8, then extract '.deleted' in a variable and use len(...)

If there's an error restoring the user in case after '.deleted' is removed, it is still not unique, throw an error and tell the admin that there is already an email registered on this name. Manually edit that and try restoring

You should not automate human decisions

Reply:
@iamareebjamal I've updated the PR. Please take a look at the updated conditions now